### DISCUSSION TITLE
pthread_mutex_lock called on a destroyed mutex.

### DISCUSSION THREAD

**Discussion by hipoom** [Tencent/libpag#2476] [General]
![截屏2024-09-18 上午10 44 21](https://github.com/user-attachments/assets/c8861045-56ea-4dca-8ee5-4e835ca9f73e)

0
#00 pc 0000000000089fcc /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) [arm64-v8a::a85ef5fb74e0282813cd1edb10854577]
1
#01 pc 000000000008bd70 /apex/com.android.runtime/lib64/bionic/libc.so (__fortify_fatal(char const*, ...)+124) [arm64-v8a::a85ef5fb74e0282813cd1edb10854577]
2
#02 pc 00000000000ece4c /apex/com.android.runtime/lib64/bionic/libc.so (HandleUsingDestroyedMutex(pthread_mutex_t*, char const*)+60) [arm64-v8a::a85ef5fb74e0282813cd1edb10854577]
3
#03 pc 00000000000ecc84 /apex/com.android.runtime/lib64/bionic/libc.so (pthread_mutex_lock+156) [arm64-v8a::a85ef5fb74e0282813cd1edb10854577]
4
#04 pc 000000000023029c /data/app/~~9OwQPWZOmbYiPyN9EEXHyA==/com.tencent.map-N27c9AvTbVBZWWbDzWkJmg==/lib/arm64/libpag.so [arm64-v8a::e939ad5f68d8f43efb110377c44e73d4]
5
#05 pc 00000000000e145c /data/app/~~9OwQPWZOmbYiPyN9EEXHyA==/com.tencent.map-N27c9AvTbVBZWWbDzWkJmg==/lib/arm64/libpag.so (pag::PAGLayer::duration() const+68) [arm64-v8a::e939ad5f68d8f43efb110377c44e73d4]
6
#06 pc 000000000010aedc /data/app/~~9OwQPWZOmbYiPyN9EEXHyA==/com.tencent.map-N27c9AvTbVBZWWbDzWkJmg==/lib/arm64/libpag.so (Java_org_libpag_PAGLayer_duration+68) [arm64-v8a::e939ad5f68d8f43efb110377c44e73d4]
7
#07 pc 000000000035d3dc /data/app/~~9OwQPWZOmbYiPyN9EEXHyA==/com.tencent.map-N27c9AvTbVBZWWbDzWkJmg==/oat/arm64/base.odex [arm64-v8a::ddb9d9ce2d761b9b02b0809f9d26ed14]
8
java:
9
org.libpag.PAGLayer.long duration()(PAGLayer.java:0)
10
org.libpag.PAGImageView.void onAnimationUpdate(org.libpag.PAGAnimator)(PAGImageView.java:5)
11
org.libpag.PAGAnimator.void onAnimationUpdate()(PAGAnimator.java:3)

**Comment by hipoom** [comment]
好像是在调用 pag::PAGLayer::duration() 时有异常，现在这个问题每天出现的次数还非常大。 更多信息可以企微联系我，hipengzheng.

**Reply by kevingpqi123** [reply]
公司内外  Github Discussion 是唯一的对接渠道，我们不是实时对接的，反馈问题建议将问题补充完整，如完整的堆栈信息，尤其是涉及到锁相关，尽量包含完整的，还有出现问题的版本号、使用方式等信息
